build-system/git.js

```diff
@@ -27,7 +27,8 @@ const {getStdout} = require('./exec');
  * @return {!Array<string>}
  */
 exports.gitDiffNameOnlyMaster = function() {
-  return getStdout('git diff --name-only master').trim().split('\n');
+  const branch = gitBranchName();
+  return getStdout(`git diff --name-only master...${branch}`).trim().split('\n');
```

Comment:
This function is used in the `--local-changes` mode, so adding the branch name will no longer pick up uncommitted local changes.

I'm still curious as to where those extra changes came from, since there should never be any uncommitted local changes on Travis. Curious: In the failing case, did `master` on your fork diverge from the branch point of the PR?

Reply:
I believe I found a solution that works for uncommitted changes too:

```
git merge-base master HEAD
```
... will return the branch point. Use this instead of `master` in all these commands, and drop the `...${branch}` suffix so uncommitted changes are included too.

Happy to send you a PR that fixes all other instances if you'd like.

Reply:
Sent you #15504